### PR TITLE
feat: custom home tab

### PR DESCRIPTION
**Pull request by cloudflypeng**
this conmit need Cantonese and Traditional Chinese support

### CODE REVIEW

**Comment by hakadao** [comment]
Thxs. However, remember to point out what language you cannot translate in your PR, this will help us to maintain the untranslated language. 
https://github.com/hakadao/BewlyBewly?tab=readme-ov-file#i18n

And it's possible that the default tab configuration may not be incorporated into the settings
https://github.com/hakadao/BewlyBewly/issues/269#issuecomment-1953594171

**Comment by cloudflypeng** [comment]
> default tab configuration

the default tab configuration has been deleted

**Comment by hakadao** [comment]
十分感謝，今晚我會詳細看看，不過記得在pr description裏面說一下你有哪些語言沒有翻譯，方便日後維護

**Comment by hakadao** [comment]
這裡學dock一樣加上reset和描述文字先吧
image

File: src/_locales/cmn-CN.yml [hunk 1]
```diff
@@ -45,6 +45,7 @@ settings:
   group_search_bar: 搜索栏
   group_recommendation_mode: 推荐模式
   group_search_page_mode: 搜索页模式
+  group_home_tab: 首页tab栏配置
```
Comment (hakadao) [inline src/_locales/cmn-CN.yml:48]: 直接叫首頁tab

Comment (hakadao) [inline src/_locales/cmn-CN.yml:48]: 直接叫 `首页标签栏` 吧還是
group_home_tabs: 首页标签栏

File: src/_locales/cmn-CN.yml [hunk 2]
```diff
@@ -123,6 +124,8 @@ settings:
   settings_shared_with_the_search_page: 与搜索页共用的配置
   settings_shared_with_the_search_page_desc: 这些设置都与搜索页共用
   search_page_mode_wallpaper_fixed: 将搜索页模式的壁纸固定
+  home_tab_order: Tab展示顺序
```
Comment (hakadao) [inline src/_locales/cmn-CN.yml:127]: 改成 home_tabs_adjustment:  标签栏调整

File: src/_locales/cmn-CN.yml [hunk 3]
```diff
@@ -123,6 +124,8 @@ settings:
...
+  home_tab_order_desc: 第一个激活的 Tab 项是默认界面
```
Comment (hakadao) [inline src/_locales/cmn-CN.yml:128]: home_tabs_adjustment_desc: 第一个激活的 Tab 项是默认界面

File: src/_locales/en.yml [hunk 4]
```diff
@@ -45,6 +45,7 @@ settings:
   group_search_bar: Search Bar
   group_recommendation_mode: Recommendation Mode
   group_search_page_mode: Search Page Mode
+  group_home_tab: Tab config
```
Comment (hakadao) [inline src/_locales/en.yml:48]: group_home_tabs: Home Tabs

File: src/_locales/en.yml [hunk 5]
```diff
@@ -124,6 +125,8 @@ settings:
   settings_shared_with_the_search_page: Settings shared with the search page
   settings_shared_with_the_search_page_desc: Those settings are used in common with the search page
   search_page_mode_wallpaper_fixed: Make the wallpaper of the search page mode fixed
+  home_tab_order: Tab Order
+  home_tab_order_desc: The first activated tab is the default page
```
Comment (hakadao) [inline src/_locales/en.yml:129]: home_tabs_adjustment: Tabs adjustment
home_tabs_adjustment_desc: The first activated tab is the default page

**Review by hakadao — CHANGES_REQUESTED** [review]

**Comment by cloudflypeng** [comment]
i fix it , but there is one thing i dont no how to do in this project
the i18n use t() but it just can use in setup or the template
so i copy the tabs list 

```typescript
const homeTabs = computed((): HomeTab[] => {
  return [
    {
      label: t('home.for_you'),
      value: HomeSubPage.ForYou,
    },
    {
      label: t('home.following'),
      value: HomeSubPage.Following,
    },
    {
      label: t('home.subscribed_series'),
      value: HomeSubPage.SubscribedSeries,
    },
    {
      label: t('home.trending'),
      value: HomeSubPage.Trending,
    },
    {
      label: t('home.ranking'),
      value: HomeSubPage.Ranking,
    },
  ]
})
```

and now this list is in two file, how can i writer to defined it in one place

**Comment by hakadao** [comment]
> i fix it , but there is one thing i dont no how to do in this project
> the i18n use t() but it just can use in setup or the template
> so i copy the tabs list

u can just store the i18n key like this 
image
when you wanna use it
image

btw im not sure if i should put it into a new Consts.ts file to store all these constants, or maybe put it into the ~/store/ (?)🤔

anyway, thank u for providing the new method to compare the stored object list and the default object list

**Pull request merged**